File: app/src/main/java/de/htwdd/htwdresden/MensaCreditActivity.java
package de.htwdd.htwdresden;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.TextView;

import com.heinrichreimer.canteenbalance.cardreader.CardBalance;

public class MensaCreditActivity extends AppCompatActivity {
    private static final String MENSA_CREDIT = "MENSA_CREDIT";
    private static final String MENSA_LAST_TRANSACTION = "MENSA_LAST_TRANSACTION";
    private static final String MENSA_HAS_MENSA_LAST_TRANSACTION = "MENSA_HAS_MENSA_LAST_TRANSACTION";

    public static Bundle setArguments(final @NonNull CardBalance cardBalance) {
        final Bundle bundle = new Bundle();
        bundle.putString(MENSA_CREDIT, cardBalance.getBalance());
        bundle.putString(MENSA_LAST_TRANSACTION, cardBalance.getLastTransaction());
        bundle.putBoolean(MENSA_HAS_MENSA_LAST_TRANSACTION, cardBalance.isLastTransactionSupported());
        return bundle;
    }

    @Override
    protected void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_mensa_dialog_credit);

        // Schließen Button
        findViewById(R.id.mensa_button_close).setOnClickListener(view -> finish());
        // Guthaben anzeigen
        showCredit(getIntent().getExtras());
    }

    @Override
    protected void onNewIntent(final Intent intent) {
        showCredit(intent.getExtras());
        super.onNewIntent(intent);
    }


   private void showCredit(@Nullable final Bundle bundle) {
        if (bundle == null) {
            return;
        }

        ((TextView) findViewById(R.id.mensa_credit_current)).setText(getString(R.string.mensa_euro_string, bundle.getString(MENSA_CREDIT, "0,00")));
        if (bundle.getBoolean(MENSA_HAS_MENSA_LAST_TRANSACTION, false)) {
            ((TextView) findViewById(R.id.mensa_last_transaction)).setText(getString(R.string.mensa_euro_string, bundle.getString(MENSA_LAST_TRANSACTION, "0,00")));
        } else {
            ((TextView) findViewById(R.id.mensa_last_transaction)).setText("-");
        }
    }
}

File: app/src/main/java/de/htwdd/htwdresden/ui/views/activities/MensaCreditActivity.kt
package de.htwdd.htwdresden.ui.views.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.os.bundleOf
import com.heinrichreimer.canteenbalance.cardreader.CardBalance
import de.htwdd.htwdresden.R
import de.htwdd.htwdresden.utils.extensions.click
import de.htwdd.htwdresden.utils.extensions.toggle
import kotlinx.android.synthetic.main.activity_mensa_credit.*

class MensaCreditActivity: AppCompatActivity() {

    companion object {
        const val ARG_CREDIT                = "CREDIT"
        const val ARG_LAST_TRANSACTION      = "LAST_TRANSACTION"
        const val ARG_HAS_LAST_TRANSACTION  = "HAS_LAST_TRANSACTION"

        fun setBundle(cardBalance: CardBalance) = bundleOf(
            ARG_CREDIT to cardBalance.balance,
            ARG_LAST_TRANSACTION to cardBalance.lastTransaction,
            ARG_HAS_LAST_TRANSACTION to cardBalance.isLastTransactionSupported
        )
    }

    private val credit by lazy {
        resources.getString(R.string.mensa_euro, intent.extras?.getString(ARG_CREDIT, "0,00") ?: "0,00")
    }
    private val hasLastTransaction by lazy {
        intent.extras?.getBoolean(ARG_HAS_LAST_TRANSACTION, false) == true
                && (intent.extras?.getString(ARG_LAST_TRANSACTION, "0,00") ?: "0,00") != "0,00"
    }
    private val lastTransaction by lazy {
        if (hasLastTransaction) {
            resources.getString(R.string.mensa_euro, intent.extras?.getString(ARG_LAST_TRANSACTION, "0,00") ?: "0,00")
        } else {
            ""
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mensa_credit)
        setup()
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        setup()
    }

    private fun setup() {
        btnClose.click { finish() }
        tvCredit.text = credit
        tvLastTransactionLabel.toggle(hasLastTransaction)
        tvLastTransaction.apply {
            toggle(hasLastTransaction)
            text = lastTransaction
        }
    }
}